extensions/carbon-touchpoint/src/main/java/org/wso2/carbon/extensions/touchpoint/actions/ChmodAction.java

```diff
@@ -0,0 +1,161 @@
+/*
+* Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.wso2.carbon.extensions.touchpoint.actions;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.equinox.p2.engine.spi.ProvisioningAction;
+import org.wso2.carbon.extensions.touchpoint.utils.Constants;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Map;
+
+/**
+ * Custom action for Chmod.
+ */
+public class ChmodAction extends ProvisioningAction {
+    private static final boolean WINDOWS = java.io.File.separatorChar == '\\';
+
+    /**
+     * {@inheritDoc}
+     * @param parameters
+     * @return
+     */
+    @Override
+    public IStatus execute(Map<String, Object> parameters) {
+        Object absoluteFiles = parameters.get(Constants.PARM_ABSOLUTE_FILES); //String or String[]
+        String targetDir = (String) parameters.get(Constants.PARM_TARGET_DIR);
+        String targetFile = (String) parameters.get(Constants.PARM_TARGET_FILE);
+        String profile = parameters.get(Constants.PROFILE).toString();
+
+        if (targetFile != null && absoluteFiles != null) {
+            return new Status(IStatus.ERROR, Constants.PLUGIN_ID, "Both Target file and absolute files can't be set");
+        }
+        if (targetDir != null && targetFile == null) {
+            return new Status(IStatus.ERROR, Constants.PLUGIN_ID, "Target file is not set");
+        }
+        if (targetDir == null && targetFile != null) {
+            return new Status(IStatus.ERROR, Constants.PLUGIN_ID, "Target directory is not set");
+        }
+
+        String permissions = (String) parameters.get(Constants.PARM_PERMISSIONS);
+        if (permissions == null) {
+            return new Status(IStatus.ERROR, Constants.PLUGIN_ID, "Permission is not set");
+        }
+
```

Comment:
Let's add an inline comment here saying, this code segment is specific to WSO2 runtime retrieval related logic.